src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/CompletionListOptimizer.cs

```diff
@@ -1,10 +1,11 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT license. See License.txt in the project root for license information.
 
+extern alias RLSP;
+
 using System.Collections.Generic;
 using System.Linq;
-using Microsoft.VisualStudio.LanguageServer.Protocol;
-using AliasedVSCommitCharacters = Microsoft.VisualStudio.LanguageServer.Protocol.SumType<string[], Microsoft.VisualStudio.LanguageServer.Protocol.VSInternalCommitCharacter[]>;
+using AliasedVSCommitCharacters = RLSP::Roslyn.LanguageServer.Protocol.SumType<string[], RLSP::Roslyn.LanguageServer.Protocol.VSInternalCommitCharacter[]>;
```

Comment:
Is it possible to add a special namespace alias in GlobalUsings.cs to avoid needing the RLSP extern alias? I'm imagining that "global using RoslynLsp = RSLP::Roslyn.LanguageServer.Protocol` could be declared in GlobalUsings.cs and then you can write `using AliasedVSCommitCharacters = RoslynLsp.SumType<string[], RoslynLsp.VSInternalCommitCharacter[]>;`

Note: I suspect that might not be possible, but I wanted to check. :smile:

Reply:
Cyrus suggested a similar thing originally, and I tried to just have `global using Roslyn = RLSP::Roslyn;`, so that we wouldn't need a global using and also not need `extern alias` in every .cs file, but sadly it doesn't work in `using` directives. Not the first time the "alias any type" feature has not lived up to its name 😛

It does work outside of using directives, but there are exactly 0 use cases of that in this PR.

Reply:
Yeah, I suspected that might not work. It's not a common scenario, and I imagine it might even have enough nuance to elevate it to a separate feature. The way that "global using" directives were implemented is that it's as if all global usings are pasted at the top of the user's file. They don't create a new scope, which would probably be necessary for this to work.